pkg/cmd/taskrun/log_reader.go

```diff
@@ -227,3 +235,45 @@ func getSteps(pod *corev1.Pod) []*step {
 
 	return steps
 }
+
+// reading of logs should wait untill the name of pod is
+// updates in the status. Open a watch channel on run
+// and keep checking the status until the pod name updatea
+// or the timeout is reached
+func (lr *LogReader) waitUntilPodNameAvailable(timeout time.Duration) (*v1alpha1.TaskRun, error) {
+	var first = true
+	opts := metav1.ListOptions{
+		FieldSelector: fields.OneTermEqualSelector("metadata.name", lr.Run).String(),
+	}
+	tkn := lr.Clients.Tekton
+	run, err := tkn.TektonV1alpha1().TaskRuns(lr.Ns).Get(lr.Run, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	if run.Status.PodName != "" {
+		return run, nil
+	}
```

Comment:
How about just using a watch only (which is already there) and dropping this get call? So it could avoid one more get call.

Reply:
Assume the case in which Pod name has already been updated and there is no event happen in the next 10 secs.

Reply:
Watch receives a single event, even if nothing is changed. You can try out a [simple example](https://github.com/hrishin/podset-operator/blob/step-1/main.go#L46). If that doesn't work then you're right, will keep it as it is :+1: